commit 7b1c6fc69796d7bb3883a578a0798a733eb0de06
Author: Ravindu Deshan <droidcasts@protonmail.com>
Date:   Sat Mar 29 00:45:51 2025 +0530

    kernel/module.c: Ignore symbols crc check
    
    Force Load LKMs
    
    mmc: disable CRC check for better performance
    Signed-off-by: Ravindu Deshan <droidcasts@protonmail.com>

diff --git a/arch/arm64/configs/custom.config b/arch/arm64/configs/custom.config
index 35bc9456b..8b7656cf0 100755
--- a/arch/arm64/configs/custom.config
+++ b/arch/arm64/configs/custom.config
@@ -6,3 +6,14 @@ CONFIG_SECURITY_DEFEX=n
 CONFIG_DEBUG_KERNEL=y
 CONFIG_KALLSYMS=y
 CONFIG_KALLSYMS_ALL=y
+
+#Force Load Kernel Modules
+CONFIG_MODULES=y
+CONFIG_MODULE_FORCE_LOAD=y
+CONFIG_MODULE_UNLOAD=y
+CONFIG_MODULE_FORCE_UNLOAD=y
+CONFIG_MODVERSIONS=y
+CONFIG_MODULE_SRCVERSION_ALL=n
+CONFIG_MODULE_SIG=n
+CONFIG_MODULE_COMPRESS=n
+CONFIG_TRIM_UNUSED_KSYMS=n
\ No newline at end of file
diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index 659b5ff09..70ef82cea 100755
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -58,7 +58,7 @@ static const unsigned freqs[] = { 400000, 300000, 200000, 100000 };
  * performance cost, and for other reasons may not always be desired.
  * So we allow it it to be disabled.
  */
-bool use_spi_crc = 1;
+bool use_spi_crc = 0;
 module_param(use_spi_crc, bool, 0);
 
 static int mmc_schedule_delayed_work(struct delayed_work *work,
diff --git a/kernel/module.c b/kernel/module.c
index 83b459f6a..fe36dd65f 100755
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -1315,9 +1315,9 @@ static int check_version(const struct load_info *info,
 	return 1;
 
 bad_version:
-	pr_warn("%s: disagrees about version of symbol %s\n",
+	pr_warn("%s: disagrees about version of symbol %s\n, but ignore...",
 	       info->name, symname);
-	return 0;
+	return 1;
 }
 
 static inline int check_modstruct_version(const struct load_info *info,
